apps/examples/media_tash/media_tash_main.c

```diff
@@ -0,0 +1,420 @@
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <dirent.h>
+#include <string.h>
+#include <apps/shell/tash.h>
+#include <fcntl.h>
+
+#include <media/media_player.h>
+#include <media/media_recorder.h>
+#include <media/media_utils.h>
+
+#include <dm/dm_connectivity.h>
+#include <dm/dm_error.h>
+#include <sys/socket.h>
+
+#define DISCONNECT_WIFI_AND_CLOSE_SOCKET() do { dm_conn_wifi_disconnect(); disconnect_socket(&fd); } while(0)
+
+int isWiFiConnected = 0;
+
+void WiFi_Link_Up(void)
+{
+	isWiFiConnected = 1;
+	printf("\n[WiFi] Connected!\n");
+}
+
+void WiFi_Link_Down(void)
+{
+	isWiFiConnected = 0;
+	printf("\n[WiFi] Disconnected!\n");
+}
+
+int network_setup(void)
+{
+	isWiFiConnected = 0;
+
+	dm_cb_register_init();
+	dm_conn_wifi_connect(WiFi_Link_Up, WiFi_Link_Down);
+
+	while (true) {
+		if (isWiFiConnected == 1) {
+			break;
+		}
+		printf("[WiFi] Waiting Connect......\n");
+		sleep(1);
+	}
+	dm_conn_dhcp_init();
+
+	if (isWiFiConnected == 1) {
+		return 0;
+	} else {
+		return -1;
+	}
+}
+
+int connect_socket(const char *ip, int port) {
+	int fd = -1;
+	struct sockaddr_in dest;
+
+	if ((fd = socket(AF_INET, SOCK_STREAM, 0)) == -1) {
+		printf("[WiFi] Open Socket Failed!\n");
+		return -1;
+	}
+
+	memset(&dest, 0, sizeof(struct sockaddr_in));
+	dest.sin_family = PF_INET;
+	dest.sin_addr.s_addr = inet_addr(ip);
+	dest.sin_port = htons(port);
+
+	if (connect(fd, (struct sockaddr *)&dest, sizeof(struct sockaddr)) < 0) {
+		printf("[TCPCLIENT] connect fail: %d\n", errno);
+		return -1;
+	}
+
+	return fd;
+}
+
+void disconnect_socket(int *fd) {
+	if (*fd >= 0) {
+		closesocket(*fd);
+		*fd = -1;
+	}
+}
+
+int media_tash_cb(int argc, char **argv)
+{
+	char ch;
+	unsigned char vol = 125;
+
+	int fd = -1;
+	int ret = 0;
+	int pos = 0;
+	int media_len = 0;
+
+	while (true) {
+		printf("[Media Test] v0.01\n");
+		printf("-----------------\n");
+		printf("0) network_setup\n");
+		printf("1) media_play\n");
+		printf("2) media_pause_play\n");
+		printf("3) media_resume_play\n");
+		printf("4) media_stop_play\n");
+		printf("5) playing volume up\n");
+		printf("6) playing volume down\n");
+		printf("7) Forward\n");
+		printf("8) Backward\n");
+		printf("-----------------\n");
+		printf("a) media_record\n");
+		printf("b) media_record_stream\n");
+		printf("c) media_pause_record\n");
+		printf("d) media_resume_record\n");
+		printf("e) media_stop_record\n");
+		printf("f) before record(a and d), play_file_\n");
+		printf("-----------------\n");
+		printf("x) Exit\n");
+		printf(">> ");
+
+		ch = getchar();
+		printf("Instruction: %c\n", ch);
+		switch (ch) {
+		case '0': {
+			ret = network_setup();
+
+			if (ret < 0) {
+				printf("Network Error!!\n");
+				return 0;
```

Comment:
whatever you return here, this application will work.
But As you know negative value is more useful for error case.